beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/PeerSync.java

```diff
@@ -195,19 +193,19 @@ private SafeFuture<PeerSyncResult> executeSync(
                       readyForNextRequest,
                       requestContext.startSlot,
                       requestContext.count,
-                      this::importBlock);
+                      block -> importBlock(block, blobSidecarsBySlot));
 
               LOG.debug(
                   "Request {} blocks starting at {} from peer {}",
                   requestContext.count,
                   requestContext.startSlot,
                   peer.getId());
 
-              final SafeFuture<Void> blocksRequest =
-                  peer.requestBlocksByRange(
-                      requestContext.startSlot, requestContext.count, blockListener);
-
-              return SafeFuture.allOfFailFast(blocksRequest, blobSidecarsRequest)
+              return blobSidecarsRequest
+                  .thenCompose(
+                      __ ->
+                          peer.requestBlocksByRange(
+                              requestContext.startSlot, requestContext.count, blockListener))
```

Comment:
I think it's not a very safe way to pass whole mutable map along several consumers. Also I think the last consumer (importBlock) shouldn't get anything but its block only blobSidecars. Maybe it could be cleaner if we pass only related blobSidecars down from `blobSidecarsRequest`?

Reply:
Thanks, I will have a double look to see how to improve it.